packages/web-components/src/components/leaving-ibm/leaving-ibm-composite.ts

```diff
@@ -0,0 +1,127 @@
+import 'carbon-web-components/es/components/modal/modal-body';
+import 'carbon-web-components/es/components/modal/modal-header';
+import 'carbon-web-components/es/components/modal/modal-close-button';
+import 'carbon-web-components/es/components/button/button';
+import styles from './leaving-ibm.scss';
+
+export { MODAL_SIZE } from 'carbon-web-components/es/components/modal/modal.js';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Component that rendres leaving IBM component labels.
+ *
+ * @element dds-leaving-ibm-composite
+ */
+@customElement(`${ddsPrefix}-leaving-ibm-composite`)
+class DDSLeavingIbmComposite extends LitElement {
+  /**
+   * The placeholder for `setLanguage()` Redux action that will be mixed in.
+   *
+   * @internal
+   */
+  _setLanguage?: (string) => void;
+
+  /**
+   * The placeholder for `loadTranslation()` Redux action that will be mixed in.
+   *
+   * @internal
+   */
+  _loadTranslation?: (language?: string) => Promise<Translation>;
+
+  /**
+   * Leaving IBM modal copy
+   */
+  @property({ attribute: false })
+  leavingIbmCopy: LeavingIBMLabels = { LEAVING001: '', LEAVING002: '', LEAVING003: '' };
+
+  /**
+   * Leaving IBM modal button label
+   */
+  @property({ attribute: false })
+  leavingIbmButtonLabel = '';
+
+  /**
+   * The language used for query.
+   */
+  @property()
+  language?: string;
+
+  /**
+   * `true` to open the search dropdown.
+   */
+  @property({ type: Boolean, reflect: true })
+  open = false;
+
+  /**
+   * external url triggering the leaving ibm modal.
+   */
+  @property({ reflect: true })
+  url = '';
+
+  /**
+   * Extract the hostname from the url to display in the modal.
+   */
+  // eslint-disable-next-line class-methods-use-this
+  protected _getHostname(url) {
+    const { hostname } = new URL(url);
+    return hostname;
+  }
+
+  firstUpdated() {
+    const { language } = this;
+    if (language) {
+      this._setLanguage?.(language);
+    }
+    this._loadTranslation?.(language).catch(() => {}); // The error is logged in the Redux store
+  }
+
+  updated(changedProperties) {
+    if (changedProperties.has('language')) {
+      const { language } = this;
+      if (language) {
+        this._setLanguage?.(language);
+      }
+    }
+  }
+
+  render() {
+    const { open, leavingIbmCopy, leavingIbmButtonLabel, _getHostname, url } = this;
+    return html`
+      <dds-leaving-ibm-modal open="${open}" size="sm">
+        <bx-modal-header>
+          <bx-modal-close-button></bx-modal-close-button>
+          <dds-leaving-ibm-modal-heading>${leavingIbmCopy?.LEAVING001}</dds-leaving-ibm-modal-heading>
+        </bx-modal-header>
+        <bx-modal-body>
+          <p>${leavingIbmCopy?.LEAVING002}</p>
+          <p class="${prefix}--leaving-ibm__text">${leavingIbmCopy?.LEAVING003}</p>
+          <a class="${prefix}--leaving-ibm__link" href="https://www.ibm.com">${_getHostname(url)}</a>
+        </bx-modal-body>
```

Comment:
Great if we can create an inheritance of `<bx-modal-body>` as a custom element that does something like below:

```javascript
@customElement(`${ddsPrefix}-leaving-ibm-modal-body`)
class DDSLeavingIBMModalBody extends LitElement {
  ...

  render() {
    const { href } = this;
    return html`
      <p><slot></slot></p>
      <p class="${prefix}--leaving-ibm__text"><slot name="supplemental"></slot></p>
      <a class="${prefix}--leaving-ibm__link" href="${ifNonNull(href)}"><slot name="link-description">${new URL(href).hostname}</slot></a>
    `;
  }

  ...
}
```

In this way:

* We can render the content of `<dds-leaving-ibm-composite>` to light DOM (instead of shadow DOM) by relegating the CSS definition of `.bx--leaving-ibm__text`, etc. to `<dds-leaving-ibm-modal-copy>`. We render content of `<dds-*-composite>` to light DOM so the rendered markup structure represents the semantics of the content. Also, for rendering modals, we ensure the content is rendered to the end of rendering tree using [`ModalRenderMixin`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/globals/mixins/modal-render.ts) so the modal content won't be hidden by the surrounding elements. Here's an example: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/lightbox-media-viewer/lightbox-video-player-composite.ts#L96-L114
* `<dds-leaving-ibm-composite>` does not have to deal with the HTML structure of the modal body content.